product_available_by_warehouse/models/sale.py

```diff
@@ -62,7 +62,8 @@ def onchange_show_message(self, show_message, product_id):
         # update route domain
         if route_ids:
             domain = res.get('domain', {})
-            domain['route_id'] = [('id', 'in', route_ids)]
+            # TODO: uncomment until figure out a better way
+            # domain['route_id'] = [('id', 'in', route_ids)]
             res.update({'domain': domain})
```

Comment:
sorry but I do not understand this.

You are passing a blank domain and wnat's the intention of this part.

I expect a unittest here if not we will be doing error-test until every fail generate hours of support.

Please create a unittest that given the border case return the quantity expected./